packages/studio-base/src/PanelAPI/useConfigById.ts

```diff
@@ -21,15 +29,38 @@ export default function useConfigById<Config extends Record<string, unknown>>(
   panelId: string | undefined,
 ): [Config | undefined, SaveConfig<Config>] {
   const { getCurrentLayoutState, savePanelConfigs } = useCurrentLayoutActions();
+  const extensionSettings = useExtensionCatalog(getExtensionPanelSettings);
+  const sortedTopics = useMessagePipeline((state) => state.sortedTopics);
 
   const configSelector = useCallback(
     (state: DeepPartial<LayoutState>) => {
       if (panelId == undefined) {
         return undefined;
       }
-      return maybeCast<Config>(state.selectedLayout?.data?.configById?.[panelId]);
+      const stateConfig = maybeCast<Config>(state.selectedLayout?.data?.configById?.[panelId]);
+      const panelType = getPanelTypeFromId(panelId);
+      const customSettingsByTopic: Record<string, unknown> = _.merge(
+        {},
+        ...sortedTopics.map(({ name: topic, schemaName }) => {
+          if (schemaName == undefined) {
+            return {};
+          }
+          const defaultConfig = extensionSettings[panelType]?.[schemaName]?.defaultConfig;
+          if (defaultConfig == undefined) {
+            return {};
+          }
+          return {
```

Comment:
Can you please also add unit tests to cover these scenarios?

Reply:
The last ExtensionCatalogProvider test already tests that behaviour unless you want me to put it in a separate method and test it individually, but I don't think it is necessary